### PR TITLE
Fixed #25297 -- Allowed makemessages to work with {% trans %} tags that use template filters.

### PR DESCRIPTION
**Pull request by apocquet**
More details here: https://code.djangoproject.com/ticket/25297#comment:3

### CODE REVIEW

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by timgraham** [comment]
Please check the test failure (also don't forget to update the flags on the ticket so the patch appears in the review queue). I'll mark "Patch needs improvement" due to the test failure -- please uncheck after updating.

**Comment by apocquet** [comment]
Well, I don't get it. Tests work fine for me, with both python 2.7 and 3.4.

**Comment by timgraham** [comment]
You could try squashing your commits just to make sure Jenkins isn't being flaky and picking up an old commit. Besides that, you might try temporarily adding some additional debugging to help try to understand why Jenkins is working differently. I also couldn't reproduce the failures it reports.

File: django/utils/translation/trans_real.py [hunk 1]
```diff
@@ -529,7 +529,9 @@ def blankout(src, char):
 
 
 context_re = re.compile(r"""^\s+.*context\s+((?:"[^"]*?")|(?:'[^']*?'))\s*""")
-inline_re = re.compile(r"""^\s*trans\s+((?:"[^"]*?")|(?:'[^']*?'))(\s+.*context\s+((?:"[^"]*?")|(?:'[^']*?')))?\s*""")
+inline_re = re.compile(r"""^\s*trans\s+((?:"[^"]*?")|(?:'[^']*?'))"""
```
Comment (timgraham) [inline django/utils/translation/trans_real.py]: Please use hanging indent to make better use of line lengths:
```python
inline_re = re.compile(
    r"""^\s*trans\s+((?:"[^"]*?")|(?:'[^']*?'))"""
)
```

Comment (apocquet) [inline django/utils/translation/trans_real.py]: Even by removing two tabs, the regex doesn't fit on 2 lines. Besides, the current split match the 3 parts of the regex: `trans`, the optional filters, and the optional context.
Should I resplit the regex anyway? Ignore the line length (the next regex already overflows)?

Comment (timgraham) [inline django/utils/translation/trans_real.py]: You can still multiline it -- my comment is only about indentation style.

**Comment by timgraham** [comment]
By the way, you also need to rebase your branch so that it merges cleanly.

**Comment by apocquet** [comment]
I just squashed my commits, I'll rebase them now.

**Comment by apocquet** [comment]
Are you okay with this patch?

File: tests/i18n/test_extraction.py [hunk 2]
```diff
@@ -102,15 +102,28 @@ def _assertPoLocComment(self, assert_presence, po_filename, line_number, *commen
...
         if line_number is not None:
             parts.append(':%d' % line_number)
+
         needle = ''.join(parts)
         if assert_presence:
             return self.assertIn(needle, po_contents, '"%s" not found in final .po file.' % needle)
         else:
             return self.assertNotIn(needle, po_contents, '"%s" shouldn\'t be in final .po file.' % needle)
 
+    def _getTokenLineNumber(self, path, token):
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: came case names is only for assertions

File: tests/i18n/test_extraction.py [hunk 3]
```diff
@@ -122,6 +135,8 @@ def assertLocationCommentPresent(self, po_filename, line_number, *comment_parts)
         (or `#: .\dirA\dirB\foo.py:42` on Windows)
 
         None can be passed for the line_number argument to skip checking of the :42 suffix part.
+        A string token can also be pased as line_number, in which case it will be searched in
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: Please wrap docstrings at 79 chars.

File: tests/i18n/test_extraction.py [hunk 4]
```diff
@@ -221,6 +236,7 @@ def test_template_message_context_extractor(self):
         Ensure that message contexts are correctly extracted for the
         {% trans %} and {% blocktrans %} template tags.
         Refs #14806.
+        Refs #25297.
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: The ticket reference isn't really useful I think (in general we only include if there's some information in the ticket the can't be easily expressed in comments).

File: django/utils/translation/trans_real.py [hunk 5]
```diff
@@ -529,7 +529,11 @@ def blankout(src, char):
 
 
 context_re = re.compile(r"""^\s+.*context\s+((?:"[^"]*?")|(?:'[^']*?'))\s*""")
-inline_re = re.compile(r"""^\s*trans\s+((?:"[^"]*?")|(?:'[^']*?'))(\s+.*context\s+((?:"[^"]*?")|(?:'[^']*?')))?\s*""")
+inline_re = re.compile(
+    r"""^\s*trans\s+((?:"[^"]*?")|(?:'[^']*?'))"""
```
Comment (timgraham) [inline django/utils/translation/trans_real.py]: If you could add any comments to the regex to help a casual reader looking at it, that could be a nice win.

File: tests/i18n/test_extraction.py [hunk 6]
```diff
@@ -235,6 +251,11 @@ def test_template_message_context_extractor(self):
             self.assertIn('msgctxt "Special trans context #3"', po_contents)
             self.assertMsgId("Translatable literal #7c", po_contents)
 
+            # {% trans %} with a filter
+            for code in 'abcdefgh':
```
Comment (timgraham) [inline tests/i18n/test_extraction.py]: In the context of this commit it's understandable what "abcdefgh" means, but describing the purpose in a comment would be helpful to future readers looking at the code outside the context of this commit.

**Comment by timgraham** [comment]
After making those updates, I think you can squash commits and bump the ticket to "Ready for checkin", thanks.

**Comment by timgraham** [comment]
merged in e7b7f94678c632791f0323f8c61ce611275605c5, thanks!

**Pull request closed without merging**